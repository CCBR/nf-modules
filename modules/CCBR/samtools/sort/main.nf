process SAMTOOLS_SORT {
    tag { meta.id }
    label 'process_medium'

    container 'nciccbr/ccbr_ubuntu_base_20.04:v6'

    input:
    tuple val(meta), path(bam)

    output:
    tuple val(meta), path("*.bam"), path("*.bai"), emit: bam
    tuple val(meta), path("*.csi"),                emit: csi, optional: true
    path("versions.yml"),                          emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${bam.baseName}.sorted"
    if ("$bam" == "${prefix}.bam") error "Input and output names are the same, use \"task.ext.prefix\" to disambiguate!"
    """
    # current working directory is a tmpdir when 'scratch' is set
    TMP=tmp/
    mkdir \$TMP
    trap 'rm -rf "\$TMP"' EXIT

    samtools sort \\
        ${args} \\
        -@ ${task.cpus} \\
        -o ${prefix}.bam##idx##${prefix}.bam.bai \\
        -m 2G \\
        -T \$TMP \\
        --write-index \\
        ${bam}

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        samtools: \$(echo \$(samtools --version 2>&1) | sed 's/^.*samtools //; s/Using.*\$//')
    END_VERSIONS
    """

    stub:
    def prefix = task.ext.prefix ?: "${bam.baseName}.sorted"
    """
    touch ${prefix}.bam ${prefix}.bam.bai

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        samtools: \$(echo \$(samtools --version 2>&1) | sed 's/^.*samtools //; s/Using.*\$//')
    END_VERSIONS
    """
}
