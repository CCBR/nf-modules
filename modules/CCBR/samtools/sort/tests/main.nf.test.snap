{
    "test_samtools_sort": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.sorted.bam:md5,309fd06ed676e7851eda22505132f728",
                        "test.sorted.bam.bai:md5,2c02a9b519c6979eebf07e3ad40fa12e"
                    ]
                ],
                "1": [

                ],
                "2": [
                    "versions.yml:md5,71140c760cb15f7f02cc48f3ed90149e"
                ],
                "bam": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.sorted.bam:md5,2488c73846ea4d9cb67187af1c58c716",
                        "test.sorted.bam.bai:md5,2c02a9b519c6979eebf07e3ad40fa12e"
                    ]
                ],
                "csi": [

                ],
                "versions": [
                    "versions.yml:md5,71140c760cb15f7f02cc48f3ed90149e"
                ]
            }
        ],
        "timestamp": "2023-10-24T13:53:20.473892"
    },
    "test_samtools_sort_stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.sorted.bam:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "test.sorted.bam.bai:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [

                ],
                "2": [
                    "versions.yml:md5,71140c760cb15f7f02cc48f3ed90149e"
                ],
                "bam": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.sorted.bam:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "test.sorted.bam.bai:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "csi": [

                ],
                "versions": [
                    "versions.yml:md5,71140c760cb15f7f02cc48f3ed90149e"
                ]
            }
        ],
        "timestamp": "2023-10-24T13:53:25.129302"
    }
}
